indico_toolkit/auto_populate/populator.py

```diff
@@ -0,0 +1,340 @@
+        Args:
+            dataset_id (int): The dataset id of the dataset you wish to copy
+            teach_task_id (int): The teach task id of the corresponding teach task to the dataset
+            workflow_name (string): The name of the newly created workflow
+            data_column_id (str, optional): The datacolumn id of the corresponding dataset. Defaults to 'document'
+            rename_labels (dict, optional): Dictionary in format {old_label_name : new_label_name} 
+            remove_labels (list, optional): List of labels to remove from old teach task
+        Returns:
+            Workflow: a Workflow object representation of the newly created workflow
+        """
+        dataset = self.client.call(GetDataset(dataset_id))
+        (
+            old_labelset_id,
+            old_model_group_id,
+            old_target_name_map,
+        ) = self._get_teach_task_details(teach_task_id=teach_task_id)
+        # get dataset snapshot
+        export = self.client.call(
+            CreateExport(dataset_id=dataset.id, labelset_id=old_labelset_id, wait=True)
+        )
+        csv = self.client.call(DownloadExport(export.id))
+        print("Obtained snapshot")
+
+        # create workflow
+        workflow = self.structure.create_workflow(
+            name=workflow_name, dataset_id=dataset_id
+        )
+        time.sleep(2)
+        print("Created workflow")
+        old_model_group = self.client.call(
+            GetModelGroup(id=old_model_group_id, wait=True)
+        )
+        model_type = old_model_group.task_type.lower()
+        # Create new teach task
+        workflow = self.structure.add_teach_task(
+            task_name=workflow_name,
+            labelset_name=workflow_name,
+            target_names=list(old_target_name_map.keys()),
+            dataset_id=dataset.id,
+            workflow_id=workflow.id,
+            model_type=model_type,
+            data_column=data_column,
+        )
+        (
+            new_labelset_id,
+            new_model_group_id,
+            new_target_name_map,
+        ) = self._get_teach_task_details(
+            workflow.components[-1].model_group.questionnaire_id
+        )
+        labels = self._get_labels(
+            old_model_group.id,
+            new_model_group_id,
+            csv,
+            new_target_name_map,
+            rename_labels,
+            remove_labels
+        )
+        # Label new teach task
+        result = self.structure.label_teach_task(
+            label_set_id=new_labelset_id,
+            labels=[dataclasses.asdict(label) for label in labels],
+            model_group_id=new_model_group_id,
+        )
+        if result["submitLabelsV2"]["success"] == False:
+            raise ToolkitPopulationError("Error: Failed to submit labels")
+        return workflow
+
+    def _get_labels(
+        self,
+        old_model_group_id: int,
+        new_model_group_id: int,
+        export_csv: pd.DataFrame,
```

Comment:
account for differently formatted csvs (either clearly call out via docstring specifying intended order or raise exception)

Reply:
Consider expanding inputs for this function to take in more generally formatted data (such as dict {filename: labels})

Reply:
Done in method `_get_labels_by_filename`